apps/rush-lib/src/cli/utilities/npm/NpmPackage.ts

```diff
@@ -0,0 +1,219 @@
+import * as path from 'path';
+import readPackageTree = require('read-package-tree');
+import { JsonFile } from '@microsoft/node-core-library';
+
+import { IPackageJson } from '../../../index';
+import Package, {
+  IRushTempPackageJson
+} from '../Package';
+
+/**
+ * Used by the "rush link" algorithm when doing NPM package resolution.
+ */
+export interface IResolveOrCreateResult {
+  found: Package | undefined;
+  parentForCreate: Package | undefined;
+}
+
+/**
+ * The type of dependency; used by IPackageDependency.
+ */
+export enum PackageDependencyKind {
+  Normal,
+  /**
+   * The dependency was listed in the optionalDependencies section of package.json.
+   */
+  Optional,
+
+  /**
+   * The dependency should be a symlink to a project that is locally built by Rush..
+   */
+  LocalLink
+}
+
+export interface IPackageDependency {
+  /**
+   * The name of the dependency
+   */
+  name: string;
+  /**
+   * The requested version, which may be a pattern such as "^1.2.3"
+   */
+  versionRange: string;
+
+  /**
+   * The kind of dependency
+   */
+  kind: PackageDependencyKind;
+}
+
+export class NpmPackage extends Package {
```

Comment:
>NpmPackage [](start = 13, length = 10)

Can you please add test for this class?

Reply:
Very difficult to do, plus we are planning to remove this code in the near future.

---
In reply to: [152133530](https://github.com/Microsoft/web-build-tools/pull/446#discussion_r152133530) [](ancestors = 152133530)